aea/aea.py

```diff
@@ -175,6 +177,15 @@ def update(self) -> None:
         for task in self.resources.task_registry.fetch_all():
             task.execute()
         self.decision_maker.execute()
+        counter = 0
```

Comment:
I have added this temporarily. I don't think we should keep this for long. Once we have the `Filter` next week I propose that we pipe the messages straight to the `Filter` and the `Filter` can then allocate them to the correct `Handler` (see comments below as well).